http://dev.mutt.org/trac/ticket/3364
http://dev.mutt.org/trac/changeset/31881f38ca1e

Hack mutt_wstr_trunc to treat M_TREE characters as 1 cell.
Closes #3364.

Index: curs_lib.c
===================================================================
--- curs_lib.c	(revision 6010:c837ab790d2e)
+++ curs_lib.c	(revision 6035:31881f38ca1e)
@@ -884,5 +884,5 @@
   wchar_t wc;
   int w = 0, l = 0, cl;
-  size_t cw, n;
+  int cw, n;
   mbstate_t mbstate;
 
@@ -898,5 +898,11 @@
       cw = cl = 1;
     else
+    {
       cw = wcwidth (wc);
+      /* hack because M_TREE symbols aren't turned into characters
+       * until rendered by print_enriched_string (#3364) */
+      if (cw < 0 && cl == 1 && src[0] && src[0] < M_TREE_MAX)
+	cw = 1;
+    }
     if (cl + l > maxlen || cw + w > maxwid)
       break;
